src/main/java/com/google/cloud/tools/managedcloudsdk/install/Downloader.java

```diff
@@ -55,45 +57,37 @@ public void download() throws IOException, InterruptedException {
     if (Files.exists(destinationFile)) {
       throw new FileAlreadyExistsException(destinationFile.toString());
     }
-
     URLConnection connection = address.openConnection();
     connection.setRequestProperty("User-Agent", userAgentString);
 
     try (InputStream in = connection.getInputStream()) {
       long contentLength = connection.getContentLengthLong();
 
-      messageListener.message("Downloading " + address + "\n");
+      logger.info("Downloading " + address + " to " + destinationFile);
 
       try (BufferedOutputStream out =
           new BufferedOutputStream(
               Files.newOutputStream(destinationFile, StandardOpenOption.CREATE_NEW))) {
+
+        progressListener.start(
+            "Downloading " + String.valueOf(contentLength) + " bytes", contentLength);
```

Comment:
Is it worth a comment that `getContentLengthLong()` returns -1 if unknown length?

_Downloading 180345724 bytes_ is a bit ugly…  `BinaryByteUnit.format(contentLength)`  from  https://github.com/JakeWharton/byteunits?

Reply:
Added note for -1.

I'll add a bug to update formatting.